tests/unit/resource/request/request_service_dereference_collection_resources_test.go

```diff
@@ -0,0 +1,173 @@
+		Expect(testCase.expectedDereferencingResult.ContentStream).To(Equal(dereferencingResult.ContentStream))
+		Expect(testCase.expectedDereferencingResult.Metadata).To(Equal(dereferencingResult.Metadata))
+		Expect(expectedContentType).To(Equal(dereferencingResult.DereferencingMetadata.ContentType))
+		Expect(testCase.expectedDereferencingResult.DereferencingMetadata.DidProperties).To(Equal(dereferencingResult.DereferencingMetadata.DidProperties))
+		Expect(expectedContentType).To(Equal(types.ContentType(rec.Header().Get("Content-Type"))))
+	}
+},
+
+	Entry(
+		"can get collection of resource with an existent DID",
+		resourceCollectionTestCase{
+			didURL:         fmt.Sprintf("/1.0/identifiers/%s/metadata", testconstants.ExistentDid),
+			resolutionType: types.DIDJSONLD,
+			expectedDereferencingResult: &DereferencingResult{
+				DereferencingMetadata: &types.DereferencingMetadata{
+					DidProperties: types.DidProperties{
+						DidString:        testconstants.ExistentDid,
+						MethodSpecificId: testconstants.ValidIdentifier,
+						Method:           testconstants.ValidMethod,
+					},
+				},
+				ContentStream: types.NewDereferencedResourceList(
+					testconstants.ExistentDid,
+					[]*resourceTypes.Metadata{testconstants.ValidResource.Metadata},
+				),
+				Metadata: &types.ResolutionDidDocMetadata{},
+			},
+			expectedError: nil,
+		},
+	),
+
+	Entry(
+		"cannot get collection of resources with not existent DID",
+		resourceCollectionTestCase{
+			didURL:         fmt.Sprintf("/1.0/identifiers/%s/metadata", testconstants.NotExistentTestnetDid),
+			resolutionType: types.DIDJSONLD,
+			expectedDereferencingResult: &DereferencingResult{
+				DereferencingMetadata: &types.DereferencingMetadata{
+					DidProperties: types.DidProperties{
+						DidString:        testconstants.NotExistentTestnetDid,
+						MethodSpecificId: testconstants.NotExistentIdentifier,
+						Method:           testconstants.ValidMethod,
+					},
+				},
+				ContentStream: nil,
+				Metadata:      &types.ResolutionDidDocMetadata{},
+			},
+			expectedError: types.NewNotFoundError(testconstants.NotExistentTestnetDid, types.DIDJSONLD, nil, false),
+		},
+	),
+
+	Entry(
+		"cannot get collection of resources with a invalid DID",
+		resourceCollectionTestCase{
+			didURL:         fmt.Sprintf("/1.0/identifiers/%s/metadata", testconstants.InvalidDid),
+			resolutionType: types.DIDJSONLD,
+			expectedDereferencingResult: &DereferencingResult{
+				DereferencingMetadata: &types.DereferencingMetadata{
+					DidProperties: types.DidProperties{
+						DidString:        testconstants.InvalidDid,
+						MethodSpecificId: testconstants.InvalidIdentifier,
+						Method:           testconstants.InvalidMethod,
+					},
+				},
+				ContentStream: nil,
+				Metadata:      &types.ResolutionDidDocMetadata{},
+			},
+			expectedError: types.NewMethodNotSupportedError(testconstants.InvalidDid, types.DIDJSONLD, nil, false),
+		},
+	),
+
+	Entry(
+		"invalid representation",
```

Comment:
Let's rename it to something like `Cannot get Resource collection because of unsupported resolutionType`